tests/integration-tests/tests/networking/test_security_groups.py

```diff
@@ -45,21 +45,37 @@ def test_additional_sg_and_ssh_from(region, custom_security_group, pcluster_conf
     check_head_node_security_group(region, cluster, 22, ssh_from)
 
 
+@pytest.mark.parametrize("additional_security_groups", [False, True])
 @pytest.mark.usefixtures("os", "instance")
-def test_overwrite_sg(region, scheduler, custom_security_group, pcluster_config_reader, clusters_factory):
+def test_overwrite_sg(
+    region, scheduler, custom_security_group, pcluster_config_reader, clusters_factory, additional_security_groups
+):
     """Test vpc_security_group_id overwrites pcluster default sg on head and compute nodes, efs, fsx"""
-    custom_security_group_id = custom_security_group.cfn_resources["SecurityGroupResource"]
+    count_of_sgs = 2
+
+    custom_security_group_stack = custom_security_group(count_of_sgs=count_of_sgs)
+    custom_security_group_ids = []
+    for sg_indx in range(count_of_sgs):
+        custom_security_group_ids.append(custom_security_group_stack.cfn_resources[f"SecurityGroupResource{sg_indx}"])
+
     fsx_name, efs_name = "fsx", "efs"
     cluster_config = pcluster_config_reader(
-        vpc_security_group_id=custom_security_group_id, fsx_name=fsx_name, efs_name=efs_name
+        vpc_security_group_id=custom_security_group_ids[0],
+        fsx_name=fsx_name,
+        efs_name=efs_name,
+        additional_security_groups=additional_security_groups,
     )
     cluster = clusters_factory(cluster_config)
+    head_node_instance_id = cluster.head_node_instance_id
+    logging.info(f"HeadNode of the {cluster} is {head_node_instance_id}")
+
     ec2_client = boto3.client("ec2", region_name=region)
-    instances = _get_instances_by_security_group(ec2_client, custom_security_group_id)
-    logging.info("Asserting that head node and compute node has and only has the custom security group")
-    assert_that(instances).is_length(3 if scheduler == "slurm" else 2)
-    for instance in instances:
-        assert_that(instance["SecurityGroups"]).is_length(1)
+    if not additional_security_groups:
+        instances = _get_instances_by_security_group(ec2_client, custom_security_group_ids[0])
+        logging.info("Asserting that head node and compute node has and only has the custom security group")
+        assert_that(instances).is_length(3 if scheduler == "slurm" else 2)
+        for instance in instances:
+            assert_that(instance["SecurityGroups"]).is_length(1)
```

Comment:
you could avoid the `if not additional_security_groups` by using `assert_that(instance["SecurityGroups"]).is_length(number_of_sgs)` in the check.

Reply:
I dont want to call the `_get_instances_by_security_group` if not necessary which is why I am using the `not additional_security_groups` in the check